src/settings.c

```diff
@@ -365,6 +367,62 @@ int read_config(char *config_name, settings_t *settings)
     return 0;
 }
 
+int read_database(char *database_name, settings_t *settings)
+{
+    json_error_t error;
+    const char *section;
+    size_t index;
+    json_t *j_value;
+    json_t *j_entry;
+
+    json_t *database_json = json_array();
+
+    database_json = json_load_file(database_name, 0, &error);
+
+    if (database_json == NULL)
+    {
+        fprintf(stderr, "%s:%d:%d error:%s \n",
+                database_name, error.line, error.column, error.text);
+        return 1;
+    }
+
+    json_array_foreach(database_json, index, j_entry)
+    {
+        device_database_t *entry = (device_database_t*)malloc(sizeof(device_database_t));
+        entry->next = settings->coap.security;
+        settings->coap.security = entry;
+
+        json_object_foreach(j_entry, section, j_value)
+        {
+            if (strcasecmp(section, "uuid") == 0)
+            {
+                entry->uuid = (char*)json_string_value(j_value);
+            }
+            else if (strcasecmp(section, "psk") == 0)
+            {
+                size_t len = (strlen(json_string_value(j_value)) / 4) * 3;
+                entry->psk = (uint8_t*)malloc(len);
+                memset(entry->psk, 0, len);
+                base64_decode(json_string_value(j_value), entry->psk, &entry->psk_len);
+            }
+            else if (strcasecmp(section, "psk_id") == 0)
+            {
+                size_t len = (strlen(json_string_value(j_value)) / 4) * 3;
+                entry->psk_id = (uint8_t*)malloc(len);
+                memset(entry->psk_id, 0, len);
+                base64_decode(json_string_value(j_value), entry->psk_id, &entry->psk_id_len);
+            }
+            else
+            {
+                fprintf(stdout, "Unrecognised configuration file section: %s\n", section);
+            }
+        }
+//      add error on missing section?
```

Comment:
Yes, verify that j_entry has all required fields. Return error (fail-fast) if it's invalid.